docs/DEVICE-CONNECTIVITY.md

```diff
@@ -137,3 +137,239 @@ The device reports the status of all of the device connectivity using [SystemAda
+wwan config, status and metrics with NIM simply by reading/writing files located under `/run/wwan`
+directory:
+
+- `config.json` is published by NIM and may contain configuration for a cellular modem.
+- `status.json` and `metrics.json` are published by wwan microservice to inform NIM about the current
+ cellular connectivity status and to publish packet/byte counters, respectively.
+- `resolv.conf/<interface-name>.dhcp` contains the list of nameservers that should be used with
+ a given wwan interface (published by wwan to NIM and further via `DeviceNetworkStatus` to other
+ microservices, like downloader)
+
+### Netdump and Nettrace
+
+EVE performs all communication with the controller over the HTTP protocol using the [http.Client](https://pkg.go.dev/net/http#Client)
+for Go. Additionally, the same protocol and the client are typically also used to download EVE/app
+images from data stores. Given all that, it is useful to be able to obtain diagnostics from HTTP
+requests processing as done by `http.Client` and use it for connectivity troubleshooting.
+For exactly this purpose we implemented [nettrace package](../libs/nettrace/README.md),
+which internally wraps and injects hooks into `http.Client` to monitor and record a summary of
+all important network operations that happen behind the scenes during request processing at different
+layers of the network stacks, denoted as "network traces" (or just collectively denoted as "network trace"
+to avoid overusing the plural form). This would include all successful and also failed TCP and TLS
+handshakes, DNS queries, reused connections, HTTP redirects, socked read/write operations, etc.
+All of this is typically hidden behind the `http.Client` implementation and only a single error
+value with limited information is available when a request fails.
+Moreover, `nettrace` allows to capture conntrack entries as well as all packets of traced TCP
+connections and UDP "exchanges" for more in-depth analysis. For example, with a conntrack entry attached,
+it is possible to tell for a given TCP/UDP traffic which ACL rule (configured by EVE as iptables rule)
+was applied.
+
+Network tracing comes with an additional overhead and the output can be quite large. Therefore,
+we must be careful about how often are network traces obtained and how do we publish them.
+For example, logging network trace as a single message is not an option. Instead, EVE publishes
+network traces inside Tar/GZip archives, labeled as "netdumps" and stored persistently under
+`/persist/netdump` directory.
+This is done by the [netdump package](../pkg/pillar/netdump), which additionally adds some more files
+into each archive to capture the config/state of the device connectivity at the moment of the publication.
+All this information combined allows to troubleshoot a connectivity issue (between device
+and the controller or a data-store) even after it is no longer reproducible.
```

Comment:
Are we prioritizing keeping netdump for the failed cases over the successes? If not, there is an assumption that the netdumps be copied off the device sooner rather than later whichw e should state here.

Should we add a pointer to edgeview as the mechanism to get the dumps? (I think you said the USB stick also works?)

Reply:
Success and Fail cases are published into separate topics (mentioned below in the md document). Since every topic is rotated separately (i.e. we will have 10 fail netdumps and 10 success netdumps at max by default - per nim, zedagent and downloader), meaning that only newer failure can cause a removal of a netdump with an older failure.

How to get those netdumps is mentioned in the last paragraph (scp, edgeview, USB stick, remote console).